mlir/test/Integration/Dialect/SparseTensor/CPU/sparse_pack.mlir

```diff
@@ -279,6 +279,31 @@ module {
     %si = tensor.extract %li[] : tensor<i64>
     vector.print %si : i64
 
+    // TODO: This check is no longer needed once the codegen path uses the
+    // buffer deallocation pass. "dealloc_tensor" turn into a no-op in the
+    // codegen path.
+    %has_runtime = sparse_tensor.has_runtime_library
+    scf.if %has_runtime {
+      // sparse_tensor.assemble copies buffers when running with the runtime
+      // library. Deallocations are needed not needed when running in codgen
```

Comment:
Not needed?

Reply:
Yes that's right. The tensors that are fed to `%s4` etc. are `arith.constant`, which bufferize to `memref.get_global`. Memref globals do not need deallocations.

Reply:
Thanks for the explanation! The comment actually said `Deallocations are needed not needed`. Needed or not needed, that is the question ;)

Reply:
ohhhh